autoscheduler/frontend/src/components/SchedulingPage/OptionsCard/OptionsCard.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import {
+  Button, Checkbox, ListItem, ListItemIcon, ListItemText,
+} from '@material-ui/core';
+import { useDispatch, useSelector } from 'react-redux';
+import GenericCard from '../../GenericCard/GenericCard';
+import * as styles from './OptionsCard.css';
+import { replaceSchedules } from '../../../redux/actions/schedules';
+import selectSchedule from '../../../redux/actions/selectedSchedule';
+import Meeting from '../../../types/Meeting';
+// DEBUG
+import fetch from './generateSchedulesMock';
+import { RootState } from '../../../redux/reducer';
+import { CourseCardArray } from '../../../types/CourseCardOptions';
+import Availability from '../../../types/Availability';
+import { formatTime } from '../../../timeUtil';
+
+const OptionsCard: React.FC = () => {
+  const [includeFull, setIncludeFull] = React.useState(false);
+  const courseCards = useSelector<RootState, CourseCardArray>((state) => state.courseCards);
+  const avsList = useSelector<RootState, Availability[]>((state) => state.availability);
+  const dispatch = useDispatch();
+  const fetchSchedules = React.useCallback(() => {
+    // make courses object
+    const courses = [];
+    for (let i = 0; i < courseCards.numCardsCreated; i++) {
+      if (courseCards[i] && courseCards[i].course) {
+        courses.push({
+          subject: courseCards[i].course.split(' ')[0],
+          courseNum: courseCards[i].course.split(' ')[1],
+          sections: courseCards[i].sections,
+          honors: courseCards[i].honors,
+          web: courseCards[i].web,
+        });
+      }
+    }
+    // make availabilities object
+    const availabilities = avsList.map((avl) => ({
+      startTime: formatTime(avl.startTimeHours, avl.startTimeMinutes).replace(':', ''),
+      endTime: formatTime(avl.endTimeHours, avl.endTimeMinutes).replace(':', ''),
+      day: avl.dayOfWeek,
+    }));
+
+    fetch('/api/scheduling/generate', {
+      body: JSON.stringify({
+        term: '202011',
+        courses,
+        availabilities,
+      }),
```

Comment:
```suggestion
    fetch('/api/scheduling/generate', {
      method: 'POST',
      body: JSON.stringify({
        term: '202011',
        includeFull,
        courses,
        availabilities,
      }),
```

It won't actually affect anything right now (at least I don't think it will), but since it'll be a POST request we might as well start with it set up like one.

Should also pass along `includeFull` with the request since it'll be part of it when the route is actually implemented (not sure of the correct syntax for it but you get the point)